cpp/cmake/thirdparty/get_cucollections.cmake

```diff
@@ -15,7 +15,7 @@
 # This function finds cuCollections and performs any additional configuration.
 function(find_and_configure_cucollections)
   include(${rapids-cmake-dir}/cpm/cuco.cmake)
-  rapids_cpm_cuco(BUILD_EXPORT_SET cudf-exports)
+  rapids_cpm_cuco(BUILD_EXPORT_SET cudf-exports INSTALL_EXPORT_SET cudf-exports)
```

Comment:
This makes sense, but I am confused why the below `rapids_export_package` is not having the same effect. The error is definitely occurring for you when `BUILD_SHARED_LIBS=OFF`, right? So the `rapids_export_package` call should be happening?

Reply:
Verified via CMakeCache file that BUILD_SHARED_LIBS is need OFF.  No idea why the original code isn't installing the cuco cmake files as it should.

Reply:
I'm fine with this change since it seems like we should be doing it anyway (we should probably remove the manual `rapids_export_package` logic below since it's redundant now, at least in theory), but I'm just going to play with the build a little to try to understand why this is happening to make sure that we have the right fix and aren't breaking the shared lib use case in some surprising way.

Reply:
This should only be added when `BUILD_SHARED_LIBS` is a false value.

Reply:
If the cuco cmake files are only installed when building static libraries, how does `rapids_find_package(cudf)` work in the shared library case?  Or is cuco not listed as a dependency of libcudf in that case?

Reply:
It isn't listed as a dependency in those cases as the headers aren't exposed as part of the public API. 

We only need to expose it for static linking.